### PR TITLE
[move prover] build a GlobalEnv from bytecodes

### PR DESCRIPTION
**Pull request by sblackshear**
In order to deploy the read/write set analysis, the Diem adapter will need to build a GlobalEnv from raw bytecodes stored on-chain. This PR makes that possible by adding a new builder for the GlobalEnv that needs only a list of modules topologically sorted by the module dependency relation.

To test the change, I run the bytecode builder on the modules extracted from each GlobalEnv in the existing test harness + do some sanity checks to ensure that both environments contain the same code.

### CODE REVIEW

File: language/vm/src/normalized.rs [hunk 1]
```diff
@@ -108,6 +108,20 @@ impl Module {
             public_functions,
         }
     }
+
+    /// Return the private functions of module `m`
```
Comment (bob-wilson) [inline language/vm/src/normalized.rs:112]: I don't see this new function being used anywhere. Same for the is_copy function below. Should those be part of a separate PR?

Comment (sblackshear) [reply language/vm/src/normalized.rs:112]: My bad; I neglected to check in the relevant file.

File: language/move-model/src/builder/bytecode_builder.rs [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+            let struct_name = QualifiedSymbol {
+                module_name: module_name.clone(),
+                symbol,
+            };
+            let type_params = make_type_params(&mut builder, &s.type_parameters);
+            let is_resource = !s.is_copy();
+
+            // TODO: this will crash if a struct depends on types declared in the same module.
+            // Sort normalized.structs by dependency relation so this is safe.
+            /*let mut fields = BTreeMap::new();
```
Comment (wrwg) [inline language/move-model/src/builder/bytecode_builder.rs:43]: The full builder first enters all structs and then adds their fields. Can't you do the same here? The current solution seems to be rather incomplete. Also it appears sorting would assume acyclic (non-recursive) structs. Even if that is the case right now I would not encode this as an assumption here for future resilience.

See also comment on make_type for another possible solution.

File: language/move-model/src/builder/bytecode_builder.rs [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                .map(|t| make_type(builder, t))
+                .collect();
+            let menv = builder.env.find_module(&module_name).unwrap_or_else(|| {
+                panic!(
+                "Couldn't resolve module {:?}; make sure input module list is sorted in dep order",
+                module
+            )
+            });
+            let struct_id = menv
+                .find_struct_by_identifier(name.clone())
```
Comment (wrwg) [inline language/move-model/src/builder/bytecode_builder.rs:134]: Alternatively to doing two phases or topsort, you can construct a StructId from scratch without needing to have it defined already in the environment, because it is defined as StructId(Symbol), and you can assume well-formedness of bytecode.

**Review by wrwg — APPROVED** [review]
